src/rust/lib_ccxr/src/util/log.rs

```diff
@@ -0,0 +1,592 @@
+        std::process::exit(exit_cause.exit_code())
+    }
+
+    /// Log an error message. Use [`error!`] instead.
+    ///
+    /// Used for logging general errors occuring in the program.
+    pub fn log_error(&self, args: &Arguments<'a>) {
+        if self.gui_mode {
+            eprint!("###MESSAGE#")
+        } else {
+            eprint!("\rError: ")
+        }
+
+        eprintln!("{}", args);
+    }
+
+    /// Log an informational message. Use [`info!`] instead.
+    ///
+    /// Used for logging extra information about the execution of the program.
+    pub fn log_info(&self, args: &Arguments<'a>) {
+        if self.target == OutputTarget::Quiet {
+            return;
+        }
+
+        activity_header();
+
+        self.print(args);
+    }
+
+    /// Log a debug message. Use [`debug!`] instead.
+    ///
+    /// Used for logging debug messages throughout the program.
+    pub fn log_debug(&self, message_type: DebugMessageFlag, args: &Arguments<'a>) {
+        if self.debug_mask.mask().intersects(message_type) {
+            self.print(args);
+        }
+    }
+
+    /// Send a message to GUI. Use [`send_gui`] instead.
+    ///
+    /// Used for sending information related to XDS to the GUI.
+    pub fn send_gui(&self, message_type: GuiXdsMessage) {
+        if self.gui_mode {
+            match message_type {
+                GuiXdsMessage::ProgramName(program_name) => {
+                    eprintln!("###XDSPROGRAMNAME#{}", program_name)
+                }
+                GuiXdsMessage::ProgramIdNr {
+                    minute,
+                    hour,
+                    date,
+                    month,
+                } => eprintln!(
+                    "###XDSPROGRAMIDENTIFICATIONNUMBER#{}#{}#{}#{}",
+                    minute, hour, date, month
+                ),
+                GuiXdsMessage::ProgramDescription { line_num, desc } => {
+                    eprintln!("###XDSPROGRAMDESC#{}#{}", line_num, desc)
+                }
+                GuiXdsMessage::CallLetters(current_letters) => {
+                    eprintln!("###XDSNETWORKCALLLETTERS#{}", current_letters)
+                }
+            }
+        }
+    }
+
+    /// Log a hex dump which is helpful for debugging purposes.
+    /// Use [`hex_dump`] or [`hex_dump_with_start_idx`] instead.
+    ///
+    /// Setting `clear_high_bit` to true will ignore the highest bit whien displaying the
+    /// characters. This makes visual CC inspection easier since the highest bit is usually used
+    /// as a parity bit.
+    ///
+    /// The output will contain byte numbers which can be made to start from any number using
+    /// `start_idx`. This is usually `0`.
+    pub fn log_hex_dump(
+        &self,
+        message_type: DebugMessageFlag,
+        data: &[u8],
+        clear_high_bit: bool,
+        start_idx: usize,
+    ) {
+        if !self.debug_mask.mask().intersects(message_type) {
+            return;
+        }
+        println!("In if statement");
```

Comment:
Looks like it was left here while debugging😅. Please remove it

Reply:
Oh really forgot about this one :joy: